### PR TITLE
ZIOS-11203: Added support for new "managed_by" field

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

Added support for the new `managed_by` field.

### CODE REVIEW

**Comment by billypchan** [comment]
@nicorsm Ref to https://github.com/wireapp/wire-ios/pull/3184, we should branch a feature branch from release 3.27 branch and then merge to it instead of develop branch.

**Comment by nicorsm** [comment]
@billypchan sure! I'll create the `feature/readonly-profile` (and fix the tests, btw)

**Comment by alexaubry** [comment]
Will this require changes from the sync engine, or break the API on the UI?

**Comment by nicorsm** [comment]
@alexaubry I don't think so, for now, since it's just a value we need to check to enable/disable editing.

File: WireDataModel.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -45,6 +45,11 @@
 		1689FD462194A63E00A656E2 /* ZMClientMessageTests+Editing.swift in Sources */ = {isa = PBXBuildFile; fileRef = 1689FD452194A63E00A656E2 /* ZMClientMessageTests+Editing.swift */; };
 		16AD86BA1F75426C00E4C797 /* NSManagedObjectContext+NotificationContext.swift in Sources */ = {isa = PBXBuildFile; fileRef = 16AD86B91F75426C00E4C797 /* NSManagedObjectContext+NotificationContext.swift */; };
 		16C391E2214BD438003AB3AD /* MentionTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 16C391E1214BD437003AB3AD /* MentionTests.swift */; };
+<<<<<<< HEAD
+=======
```
Comment (billypchan) [inline WireDataModel.xcodeproj/project.pbxproj]: @nicorsm merge is in progress.

File: Source/Model/User/ZMUser+Internal.h [hunk 2]
```diff
@@ -50,6 +50,8 @@ extern NSString * __nonnull const ReadReceiptsEnabledKey;
 @property (nullable, nonatomic) NSData *imageMediumData;
 @property (nullable, nonatomic) NSData *imageSmallProfileData;
 
+@property (nullable, nonatomic) NSString *managedBy;
```
Comment (typfel) [inline Source/Model/User/ZMUser+Internal.h]: Shouldn't this property be annotated as a `managed` property somewhere?

Comment (typfel) [reply Source/Model/User/ZMUser+Internal.h]: I guess `@dynamic managedBy;` since it's Objective-C

File: Source/Model/User/ZMUser.m [hunk 3]
```diff
@@ -563,6 +569,11 @@ - (void)updateWithTransportData:(NSDictionary *)transportData authoritative:(BOO
         self.name = name;
     }
     
+    NSString *managedBy = [transportData optionalStringForKey:@"managed_by"];
```
Comment (typfel) [inline Source/Model/User/ZMUser.m]: A test in `ZMUserTests` checking that we parse this field would be nice.

File: Source/Model/User/ZMUser.m [hunk 4]
```diff
@@ -380,6 +383,7 @@ @implementation ZMUser (Internal)
 @dynamic showingUserAdded;
 @dynamic showingUserRemoved;
 @dynamic createdTeams;
+@dynamic managedBy;
```
Comment (nicorsm) [inline Source/Model/User/ZMUser.m]: @typfel added @dynamic here.

**Pull request merged**